float64 odo_speed